check-reserve-update-frequency/src/agent.js

```diff
@@ -1,69 +1,84 @@
 
-    // get the timestamp for the current block
-    const blockTimestamp = new BigNumber(blockEvent.block.timestamp);
-
-    // override block number so we get data from the block in question
-    const override = { blockTag: blockEvent.blockNumber };
+  // calculate the difference between the current block timestamp and the last oracle update
+  const oracleAge = blockTimestamp.minus(timestamp);
 
-    // define the promise function to run for each reserve token
-    async function checkOracleAge(tokenAddressContract) {
-      const { reserveToken, priceSourceAddress, priceSourceContract } = tokenAddressContract;
+  // return a finding if the oracle update age is greater than ORACLE_AGE_THRESHOLD_SECONDS
+  if (oracleAge.isGreaterThan(ORACLE_AGE_THRESHOLD_SECONDS)) {
+    const { developerAbbrev, protocolName, protocolAbbrev } = data;
+    return createAlert(
+      developerAbbrev, protocolName, protocolAbbrev, reserveToken, oracleAge, priceSourceAddress,
+    );
+  }
 
-      // get the timestamp from the price source contract
-      let roundData;
-      try {
-        roundData = await priceSourceContract.latestRoundData({ ...override });
-      } catch (error) {
-        return;
-      }
+  return [];
+}
 
-      // the updatedAt value is of type ethers.BigNumber
-      // ethers.BigNumber is not the same as BigNumber from bignumber.js
-      // therefore, we need to convert from ethers.BigNumber to BigNumber
-      const timestamp = new BigNumber(roundData.updatedAt.toString());
+function provideHandleBlock(data) {
+  return async function handleBlock(blockEvent) {
+    const { tokenAddressContractTuples: tokensAddressesContracts } = data;
 
-      // calculate the difference between the current block timestamp and the last oracle update
-      const oracleAge = blockTimestamp.minus(timestamp);
+    // get the timestamp for the current block
+    const blockTimestamp = new BigNumber(blockEvent.block.timestamp);
 
-      if (oracleAge.isGreaterThan(ORACLE_AGE_THRESHOLD_SECONDS)) {
-        findings.push(createAlert(reserveToken, oracleAge, priceSourceAddress));
-      }
-    }
+    // override block number so we get data from the block in question
+    const override = { blockTag: blockEvent.blockNumber };
 
     // for each reserve token, get the price source address and timestamp
     // forEach does not work with async and promises
     // attach a .catch() method to each promise to prevent any rejections from causing Promise.all
     // from failing fast
-    await Promise.all(tokensAddressesContracts.map(
-      (tokenAddressContract) => checkOracleAge(
-        tokenAddressContract,
-      ).catch((error) => console.error(error)),
-    ));
+    const findings = (await Promise.all(tokensAddressesContracts.map(
+      (tokenAddressContract) => checkOracleAge(tokenAddressContract, override, blockTimestamp, data)
+        .catch((error) => console.error(error)),
+    ))).flat();
```

Comment:
I'm not certain this will do what we want (I also think I wrote it...).  If any error is caught, the console.error() call will occur, but the resulting return value will be `undefined`, which is not what we want (calling .flat() won't remove the `undefined` elements).

I _think_ this will remedy the issue (untested suggestion):
```suggestion
    const promises = tokensAddressesContracts.map((tokenAddressContract) => {
      try {
        return checkOracleAge(tokenAddressContract, override, blockTimestamp, data);
      } catch (error) {
        console.error(error);
        return [];
      }
    });
    
    const findings = (await Promise.all(promises)).flat();
```